src/inc/apiv2/common/AbstractBaseAPI.class.php

```diff
@@ -398,7 +398,10 @@ protected static function db2json(array $feature, mixed $val): mixed
       $obj = array_map('intval', preg_split("/,/", $val, -1, PREG_SPLIT_NO_EMPTY));
     } elseif ($feature['type'] == 'dict' && $feature['subtype'] = 'bool') {
       $obj = unserialize($val);
-    } else {
+    } elseif (str_starts_with($feature['type'], 'str')) {
```

Comment:
Currently 1 of the test fails, I think its because html_entity_decode will return an empty string when you input null. So now it fails when a task color is set to null because the test expects a null value but it gets an empty string. To solve it, I think inside this if statement you can just check whether $val != null and I think then it will pass the test. The elseif statement will then become:

} elseif (str_starts_with($feauture['type'], 'str') && $val !== null)

Reply:
Ah, thanks, changed the code like you proposed.

Reply:
Just one question, do you know how to run the tests locally before doing a commit or PR?

Reply:
Yes, in order to test the new api locally you can cd to the correct folder: "cd ~/src/hashtopolis/server/ci/apiv2" and then run the following command: "pytest --exitfirst --last-failed"